src/pages/experiments/[experimentId]/data-exploration/components/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -50,42 +52,44 @@ const DiffExprCompute = (props) => {
     const numMetadata = Object.values(properties).filter((o) => o.type === 'metadataCategorical').length;
 
     if (!numMetadata) {
-      setType(ComparisonType.within);
+      dispatch(setComparisonType(ComparisonType.within));
     }
     setHasMetadata(numMetadata > 0);
 
-    // Update the hierarchies if a previously selected set were to be deleted.
-    setSelectedGroups(_.mapValues(selectedGroups, (cellSetKey) => {
-      if (!cellSetKey) {
-        return null;
-      }
-
-      const splitKey = cellSetKey.split('/')[1];
+    // If any key in group option is deleted, set option to null
+    Object.keys(comparisonGroup).forEach((type) => {
+      const groupDiff = {};
```

Comment:
My understanding is that this `groupDiff` is an object that contains the keys of all items that have to be made to null. If that is true, I suggest you rename `groupDiff` to something else that indicates better the purpose of that object, again for code readability purposes. For example, naming it `deletedKeys` would be maybe better.

Reply:
It's set as a key to make it easier to specifically add the key to the action payload by just spreading the object,